packages/core-data/src/hooks/use-entity-record.ts

```diff
@@ -66,6 +82,58 @@ export interface Options {
  * application, the page and the resolution details will be retrieved from
  * the store state using `getEntityRecord()`, or resolved if missing.
  *
+ * @example
+ * ```js
+ * import { useEntityRecord } from '@wordpress/core-data';
+ *
+ * function PageRenameForm( { id } ) {
+ *   const page = useEntityRecord( 'postType', 'page', id );
+ *   const [ title, setTitle ] = useState( () => page.record.title.rendered );
+ *
+ *   if ( page.isResolving ) {
+ *     return 'Loading...';
+ *   }
+ *
+ *   async function onRename( event ) {
+ *       event.preventDefault();
+ *       page.edit({ title });
+ *       try {
+ *           await page.save({ throwOnError: true })
+ *           createSuccessNotice( __( 'Page renamed.' ), {
+ *               type: 'snackbar',
+ *           } );
+ *       } catch(e) {
+ *            const errorMessage =
+ *                error.message && error.code !== 'unknown_error'
+ *                    ? error.message
+ *                    : __( 'An error occurred while renaming the entity.' );
```

Comment:
Translation helper function is not imported.

Reply:
Actually, we would have to list `createSuccessNotice`, `createErrorNotice`, `TextControl` and `Button`, too. It's hard to tell whether that improves the example or makes it harder to notice the important bits.